webdriver/tests/find_element_from_shadow_root/user_prompts.py

```diff
@@ -0,0 +1,129 @@
+# META: timeout=long
+
+import pytest
+
+from tests.support.asserts import (
+    assert_error,
+    assert_same_element,
+    assert_success,
+    assert_dialog_handled,
+)
+
+
+def find_element(session, shadow_id, using, value):
+    return session.transport.send(
+        "POST", "session/{session_id}/shadow/{shadow_id}/element".format(
+            session_id=session.session_id,
+            element_id=shadow_id),
+        {"using": using, "value": value})
+
+
+@pytest.fixture
+def check_user_prompt_closed_without_exception(session, create_dialog, get_shadow_page):
+    def check_user_prompt_closed_without_exception(dialog_type, retval):
+        session.url = get_shadow_page("<div><p>bar</p><div>")
+        outer_element = session.find.css("custom-shadow-element", all=False)
+        shadow_root = outer_element.shadow_root
+        inner_element = session.execute_script("return arguments[0].shadowRoot.querySelector('p')",
+                                               args=(outer_element,))
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = find_element(session, shadow_root.id, "css selector", "p")
+        value = assert_success(response)
+
+        assert_dialog_handled(session, expected_text=dialog_type, expected_retval=retval)
+
+        assert_same_element(session, value, inner_element)
+
+    return check_user_prompt_closed_without_exception
+
+
+@pytest.fixture
+def check_user_prompt_closed_with_exception(session, create_dialog, get_shadow_page):
+    def check_user_prompt_closed_with_exception(dialog_type, retval):
+        session.url = get_shadow_page("<div><p>bar</p><div>")
+        outer_element = session.find.css("custom-shadow-element", all=False)
+        shadow_root = outer_element.shadow_root
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = find_element(session, shadow_root.id, "css selector", "p")
+        assert_error(response, "unexpected alert open")
+
+        assert_dialog_handled(session, expected_text=dialog_type, expected_retval=retval)
+
+    return check_user_prompt_closed_with_exception
+
+
+@pytest.fixture
+def check_user_prompt_not_closed_but_exception(session, create_dialog, get_shadow_page):
```

Comment:
Please move the fixtures to the `conftest.py` file

Reply:
Done in 9e70d8f.

Reply:
Note that we don't do that for all the other commands. Also these fixtures are just for the user prompt tests, and don't give any value for any other test in this folder.

Reply:
@whimboo @AutomatedTester I’m happy to put them wherever anyone wants them, but let’s come to a consensus on them.

Reply:
On second thought, I'm moving them back to the test files for the sake of consistency. There are also issues with the fixtures referencing stuff that isn't imported or existing in the `conftest.py`. If we want to make a more global change to affect all user prompt test files, we can move them all in one set of changes.